internal/pkg/aws/resourcegroups/resourcegroups.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package resourcegroups provides a client to make API requests to AWS Resource Groups
+package resourcegroups
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/resourcegroups"
+)
+
+const (
+	resourceQueryType = "TAG_FILTERS_1_0"
+)
+
+type api interface {
+	SearchResources(input *resourcegroups.SearchResourcesInput) (*resourcegroups.SearchResourcesOutput, error)
+}
+
+// ResourceGroups wraps an AWS ResourceGroups client.
+type ResourceGroups struct {
+	client api
+}
+
+// ResourceGroupsClient wraps the API for ResourceGroups, for easier testing when used by other services
+type ResourceGroupsClient interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]string, error)
+}
```

Comment:
I wonder if this interface should be defined in the clients that use the `resourcegroups` package? to follow the general guideline of consumers defining interfaces over producers.

Reply:
I thought the idea was that this was a reusable interface.

Reply:
It's not a big deal. I think the reason why they recommend that is if a consumer needs another method to be provided by the interface, they'll have to update the producer's interface and increase the scope for all the other consumers.

We can tackle that later when a consumer wants to make changes to the interface.